src/renderer/src/routes/pages/Send/SendAmount.tsx

```diff
@@ -2,25 +2,85 @@ import { Fragment } from 'react'
 import { useTranslation } from 'react-i18next'
 import { TbChevronRight } from 'react-icons/tb'
 import { VscCircleFilled } from 'react-icons/vsc'
+import { TokenBalance } from '@renderer/@types/query'
+import { IAccountState } from '@renderer/@types/store'
+import { BalanceHelper } from '@renderer/helpers/BalanceHelper'
+import { FilterHelper } from '@renderer/helpers/FilterHelper'
+import { StyleHelper } from '@renderer/helpers/StyleHelper'
+import { useBalancesAndExchange } from '@renderer/hooks/useBalancesAndExchange'
+import { useModalNavigate } from '@renderer/hooks/useModalRouter'
 
-export const SendAmount = () => {
+type TAmountParams = {
+  selectedAccount?: IAccountState | null
+  selectedToken?: TokenBalance | null
+  selectedAmount: number
+  handleSelectAmount?: (amount: number) => void
+  active: boolean
+}
+
+export const SendAmount = ({
+  selectedAccount,
+  selectedToken,
+  selectedAmount,
+  handleSelectAmount,
+  active,
+}: TAmountParams) => {
   const { t } = useTranslation('pages', { keyPrefix: 'send' })
+  const { modalNavigateWrapper } = useModalNavigate()
+  const balanceExchange = useBalancesAndExchange(selectedAccount ? [selectedAccount] : [])
+
+  const calculateEstimatedFiatValue = (amount: number) => {
+    if (selectedToken) {
+      const totalEstimatedValue =
+        BalanceHelper.convertBalanceToCurrency(selectedToken, balanceExchange.exchange.data)?.convertedAmount ?? 0
+      const pricePerToken = totalEstimatedValue / selectedToken.amountNumber
```

Comment:
You can directly use the balanceExchange to get the price per token.
```suggestion
      const pricePerToken = BalanceHelper.getExchangeRatio(
        selectedToken.token.hash,
        selectedToken.blockchain,
        balanceExchange.exchange.data
      )

      return FilterHelper.currency(amount * pricePerToken)
```